src/libraries/System.Text.Json/Common/JsonSeparatorNamingPolicy.cs

```diff
@@ -16,6 +16,11 @@ internal abstract class JsonSeparatorNamingPolicy : JsonNamingPolicy
 
         public sealed override string ConvertName(string name)
```

Comment:
Why are we passing in null to something that's annotated as being non-nullable?  More fundamentally, it seems like either we should be fixing the nullable annotation to make this `string? name` if we intend to pass in null, or we should instead fix the call sites to ensure we're never passing in null.

Reply:
>> I think the current annotation is correct since it's meant to handle JSON property names/dictionary keys which cannot be null

> introduce tolerance in the new naming policy class like this PR is already doing.

If every implementation needs to tolerate nulls, then the annotation isn't correct :-)

Reply:
I've posted https://github.com/dotnet/aspnetcore/pull/47775 which should address the root cause in aspnetcore code. 

We should still update the separator class with null detection logic + tests.

Reply:
> We should still update the separator class with null detection logic + tests.

Can you elaborate what you mean by updating the separator class with null detection logic?

Reply:
Something along the lines of
```C#
if (value is null)
{
   throw new ArgumentNullException(nameof(value));
}
```

Reply:
Oh, ok, thanks.  I thought you meant still making the implementation resilient to nulls.

Reply:
Are we going to make it consistently throw across all derived classes (i.e. JsonCamelCaseNamingPolicy)? It feels odd that one implementation would throw, and another one doesn't.

Reply:
That would almost certainly break people. It probably couldn't heard if tried to make the new thing consistent with what we've already shipped.

Reply:
It is inconsistent, but the null check isn't there in the existing class because we think it's appropriate but rather because we're stuck with it.  The nullable annotation conveys nulls aren't supported.  I don't think there's enough to gain by removing the existing null check.  We just don't need to add more of them.

Reply:
I'm fine with that. What bothers me about the camel case implementation is not so much that it's tolerating null inputs but that it also returns null in that case (even though the return type is non-nullable). Do you think it is reasonable that we should have it return, say, the empty string in that case?

Reply:
Garbage in garbage out. If someone violates the constraint on the way in, it's ok to violate it on the way out. If you don't think changing the return would break anyone, we could make it empty instead. I won't lose sleep either way.